src/DotNetty.Common/Utilities/ICharSequence.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Common.Utilities
+{
+    using System.Collections.Generic;
+
+    public interface ICharSequence : IReadOnlyList<char>
```

Comment:
Probably not very big gain.  I haven't seen number of char sequence instances very big anyway. The AppendableCharSequence  is designed to be reusable local buffer along the way.  Plus, this is the first cut of basic http, there are large number of http related code using this e.g. web socket, stomp etc.